modules/docs/mdbook/default.nix

```diff
@@ -0,0 +1,50 @@
+{
+  lib,
+  config,
+  pkgs,
+  ...
+}:
+let
+  settingsFormat = pkgs.formats.toml { };
+  defaultSettings = {
+    book = {
+      language = "en";
+      multilingual = false;
+      title = "nixvim docs";
+    };
+    build.create-missing = false;
+    output.html.site-url = "/";
+    output.html.fold = {
+      enable = true;
+      level = 0;
+    };
+    preprocessor.alerts = { };
+  };
+in
+{
+  options.docs.html = {
+    site = lib.mkOption {
+      type = lib.types.package;
+      description = "HTML docs rendered by mdbook.";
+      readOnly = true;
+    };
+    settings = lib.mkOption {
+      inherit (settingsFormat) type;
+      description = ''
+        Freeform settings written to `book.toml`.
+
+        See MDBook's [Configuration](https://rust-lang.github.io/mdBook/format/configuration/index.html) docs.
+      '';
+      defaultText = defaultSettings;
+    };
+  };
+  config.docs.html = {
+    site = pkgs.callPackage ./package.nix {
+      inherit (config.docs) src;
+      inherit (config.docs.html) settings;
+      menu = config.docs.menu.src;
+      writeTOML = settingsFormat.generate;
```

Comment:
Whats the benefit of doing this instead of getting `formats` through `callPackage`?

Reply:
We also need access to `settingsFormat` in this module, not just in the package.

The module needs `settingsFormat.type`, while the package needs `settingsFormat.generate`.

We could generate the toml string in this file, however _currently_ we need to pass `settings` as an attrset so that things can be overridden, which we do when the legacy docs bundles the beta-docs.